### PR TITLE
Fix fatal error with PRB buttons

### PR DESCRIPTION
**Pull request by wjrosa**
## Changes proposed in this Pull Request:

This PR fixes a possible fatal error happening when a product cannot be found, related to PRBs' add to cart ajax endpoint. This error was caught with Grafana:
```
PHP Fatal error: Uncaught Error: Call to a member function get_type() on bool in /wordpress/plugins/woocommerce-gateway-stripe/9.2.0/includes/payment-methods/class-wc-stripe-payment-request.php:1475 Stack trace: #0 /wordpress/core/6.7.2/wp-includes/class-wp-hook.php(324): WC_Stripe_Payment_Request->ajax_add_to_cart('') #1 /wordpress/core/6.7.2/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array) #2 /wordpress/core/6.7.2/wp-includes/plugin.php(517): WP_Hook->do_action(Array) #3 /wordpress/plugins/woocommerce/9.7.1/includes/class-wc-ajax.php(98): do_action('wc_ajax_wc_stri...') #4 /wordpress/core/6.7.2/wp-includes/class-wp-hook.php(324): WC_AJAX::do_wc_ajax('') #5 /wordpress/core/6.7.2/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(false, Array) #6 /wordpress/core/6.7.2/wp-includes/plugin.php(517): WP_Hook->do_action(Array) #7 /wordpress/core/6.7.2/wp-includes/template-loader.php(13): do_action('template_redire...') #8 /wordpress/core/6.7.2/wp-blog-header.php(19): require_once('/wordpress/core...') #9 /wordpress/core/6.7.2/index.php(17): require('/wordpress/core...') #10 {main} thrown in /wordpress/plugins/woocommerce-gateway-stripe/9.2.0/includes/payment-methods/class-wc-stripe-payment-request.php on line 1475
``` 

On a regular store setup, this error should not happen since we deprecated PRBs in favor of express payment methods. But this change should at least make the code safer.

## Testing instructions

Code review should be enough. Check if the tests are still passing.

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [ ] Tested on mobile (or does not apply)

### Changelog entry

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

File: includes/payment-methods/class-wc-stripe-payment-request.php [hunk 1]
```diff
@@ -1467,35 +1467,45 @@ public function ajax_add_to_cart() {
 			define( 'WOOCOMMERCE_CART', true );
 		}
 
-		WC()->shipping->reset_shipping();
+		try {
```
Comment (wjrosa) [inline includes/payment-methods/class-wc-stripe-payment-request.php:1470]: I just wrapped the code inside the `try...catch` block, but GitHub thinks I changed more stuff here.

File: .github/workflows/php-code-coverage.yml [hunk 2]
```diff
@@ -7,6 +7,7 @@ on:
 
 permissions:
   pull-requests: write
+  contents: read
```
Comment (diegocurbelo) [inline .github/workflows/php-code-coverage.yml]: This seems unrelated to the PRB fix, right?

**Review by diegocurbelo — APPROVED** [review]
The code looks good.

I'm not sure how to check this flow (other than forcing the `throw exception`)... In any case, inverting the corresponding if it behaves as expected.

File: .github/workflows/php-code-coverage.yml (hunk 2 above)
Comment (wjrosa) [reply .github/workflows/php-code-coverage.yml]: Yeah, I tried to fix the PHP code coverage action, but it is still not working. I will revert this and try again on another PR
